WATOLINK/bci-gaming
# Implemented health bar and death animation transition

## ClickUp ticket link

[Ticket Name][Health Status Tracking](https://app.clickup.com/t/860q9n3et)



## Implementation description
Health bar details are implemented in the playerHealth script. When the player's health equals to or drops below 0, then
the death animation is triggered and the user would be directed to the death sequence.



## Steps to test
1. Enter the Health Scene, click on the Player object, then scroll down to the Player Health script. Set the health variable to whatever you like (ex: 100).
2. Run the game. You should see that the maxHealth variable would match the health variable. maxHealth would be the highest possible health. Now, drag the health variable and change the values. Note that if you go past maxHealth (ex: 100) then the health bar won't change as the bar is clamped.
3. Drag the health value below 100. As the value decreases, you should see the health deplete from right to left
4. Once the health is 0 or below 0, you should see the player character's death animation play, and then subsequently the game would switch to the Death Sequence.



## What should reviewers focus on?
Health bar UI: I located the health bar at the top left corner, would it be better if it were somewhere else?
Scene transition: Transition to Death Sequence may not flow as well. Any suggestions as to how to make the transition flow better?


## Checklist
